src/Container.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Class Container
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay;
+
+use Psr\Container\ContainerInterface;
+use WCPay\Internal\DependencyManagement\DelegateContainer\WooContainer;
+use WCPay\Internal\DependencyManagement\ExtendedContainer;
+use WCPay\Internal\DependencyManagement\ServiceProvider\PaymentsServiceProvider;
+
+/**
+ * WCPay Dependency Injection Container.
+ *
+ * Wraps the ExtendedContainer implementation to only allow public access to
+ * certain methods of the internal container.
+ *
+ * The public methods of this class should be a subset of the ExtendedContainer
+ * methods, and follow their signatures. Avoid adding other methods here.
+ *
+ * During tests, the internal ExtendedContainer is available through
+ * the `wcpay_get_test_container()` method, allowing full manipulation.
+ */
+class Container implements ContainerInterface {
+	/**
+	 * Internal container instance.
+	 *
+	 * @var ExtendedContainer
+	 */
+	private $container;
+
+	/**
+	 * Initializes the container.
+	 *
+	 * Dependencies should not be provided during runtime, but will allow
+	 * mocking during tests. This is only needed for the container.
+	 *
+	 * @param WooContainer $woo_container The delegate container for WooCommerce (Optional).
+	 */
+	public function __construct( WooContainer $woo_container = null ) {
+		$this->container = new ExtendedContainer();
+
+		// Allow the container to be used as a dependency.
+		$this->container->addShared( static::class, $this );
+
+		// Add shared services.
+		$this->load_providers();
+
+		// Allow delegating unresolved queries to the WooCommerce container.
+		$this->container->delegate( $woo_container ?? new WooContainer() );
```

Comment:
Is there any benefit to having a dependency like this `WooContainer $woo_container = null` and then still check if it is `null` and then call itself? 

So we can remove the dependency and make it simple like this one: 
```suggestion
		$this->container->delegate( new WooContainer() );
```

Reply:
@htdat I did this to allow mocking the container in tests: https://github.com/Automattic/woocommerce-payments/blob/rpp/6678-dependency-container/tests/unit/src/ContainerTest.php#L56-L57

> Is there any benefit to having a dependency like this WooContainer $woo_container = null and then still check if it is null and then call itself?

The line in question simply creates the instance if `$woo_container` isn't already set.

Reply:
I'm marking this as resolved, and merging the PR, since it's pre-approved.

Reply:
Ah, gotcha! If that is for tests, so it makes sense.